lib/storage/client.js

```diff
@@ -1182,6 +1182,251 @@ class Client extends CryptoConsumer {
     })
     return listGroups
   }
-}
 
+  /**
+   * addGroupMembers adds members to the TozStore Group given.
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   * @param {GroupMember[]} groupMembers wraps together the ClientID, Capability and Membership Key for each client being added to the group
+   *
+   *
+   * @returns {GroupMember} A response from TozStore; the group members who have been added.
+   *
+   */
+  async addGroupMembers(groupId, groupMembers = []) {
+    // Get Group Membership key for current client
+    let membershipResponse = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/membership_key`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let groupMembershipJson = await validateResponseAsJSON(membershipResponse)
+    // Get the public key for the authorizer, in order to unwrap the encrypted group key
+    // of the client adding these members
+    let publicKeysAuthResp = await this.authenticator.tokenFetch(
+      `${this.config.apiUrl}/v1/client/${groupMembershipJson.authorizer_id}/public`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let publicKeysAuthJSON = await validateResponseAsJSON(publicKeysAuthResp)
+    // Get All members Public Keys to create their membership keys
+    let clientIDs = []
+    groupMembers.forEach(member => {
+      clientIDs.push(member.clientID)
+    })
+    // Create the API object
+    let groupMembersPublicKeysRequest = {
+      client_ids: clientIDs,
+    }
+    let publicKeysResponse = await this.authenticator.tokenFetch(
+      `${this.config.apiUrl}/v1/client/public`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(groupMembersPublicKeysRequest),
+      }
+    )
+    let publicKeysJSON = await validateResponseAsJSON(publicKeysResponse)
+    // Create the new membership key
+    // for loop
+    for (var i = 0; i < groupMembers.length; i++) {
+      let newMembershipKey = await this.crypto.createGroupMembershipKey(
+        this.config.privateKey,
+        groupMembershipJson.encrypted_membership_key,
+        publicKeysJSON.clients[groupMembers[i].clientID].public_key[
+          'curve25519'
+        ],
+        publicKeysAuthJSON.public_key['curve25519']
```

Comment:
These are magic and crucial values that should be extracted out to named constant, i.e. https://github.com/tozny/e3db-clients-go/blob/master/crypto.go#L22